crates/sui-core/src/authority/authority_store.rs

```diff
@@ -764,26 +809,39 @@ impl AuthorityStore {
                     // 2. or we have some DB corruption
                     let version = shared_locks.get(id).unwrap_or_else(|| {
                         panic!(
-                        "Shared object locks should have been set. tx_digset: {:?}, obj id: {:?}",
-                        digest, id
-                    )
+                            "Shared object locks should have been set. tx_digset: {:?}, obj id: {:?}",
+                            digest, id
+                        )
                     });
-                    self.get_object_by_key(id, *version)?.unwrap_or_else(|| {
-                        panic!("All dependencies of tx {:?} should have been executed now, but Shared Object id: {}, version: {} is absent", digest, *id, *version);
-                    })
+
+                    match self.get_object_by_key(id, *version)? {
+                        Some(obj) => result.push((*kind, obj)),
+                        None => {
+                            // If the object was deleted by a concurrently certified tx then return this separately and add the digest of the current transaction 
+                            // to be used as the dependency of the next transaction that has the deleted shared object as input
+                            let epoch = epoch_store.committee().epoch;
+                            if let Some(dependency) = self.shared_object_deleted(id, epoch)? {
+                                deleted_shared_objects.push((*id, *version, dependency));
+                                self.update_deleted_shared_object_dependency(id, epoch, *digest).expect("Could not update dependency for previously deleted shared object");
```

Comment:
this is so that we update the dependencies via C => B => A instead of C => A , B => A where A is deletion and B and C are mutations. is there a way to do this update here safely?

Reply:
I've changed it to write to the existing batch in the update_state fn